### PR TITLE
tests: don't have cargo install test tools

### PR DESCRIPTION
**Pull request by martinvonz**
I noticed that `cargo install --path .` will install the new
`fake-editor` and `fake-diff-editor` binaries. We don't want that, so
let's move them into a separate crate. The tests still find them
without needing any modification.

# Checklist

- [x] I have made relevant updates to `CHANGELOG.md`

### CODE REVIEW

**Review by arxanas — COMMENTED** [review]
FWIW git-branchless puts helper binaries behind a feature flag which isn't enabled by default:

https://github.com/arxanas/git-branchless/blob/3595f3d151dffb974f7f6fddb500c79ee25a0431/git-branchless-lib/Cargo.toml#L25-L36

Not that it makes a difference either way.

**Comment by martinvonz** [comment]
> FWIW git-branchless puts helper binaries behind a feature flag which isn't enabled by default:
> 
> https://github.com/arxanas/git-branchless/blob/3595f3d151dffb974f7f6fddb500c79ee25a0431/git-branchless-lib/Cargo.toml#L25-L36

What enables the `integration-test-bin` feature?

**Comment by arxanas** [comment]
@martinvonz Me manually passing `--feature integration-test-bin` 😊

I guess what we're doing is different. Your binaries are supporting the relevant tests, but my binaries *are* the tests.

**Review by durin42 — APPROVED** [review]

**Comment by martinvonz** [comment]
> @martinvonz Me manually passing `--feature integration-test-bin` 😊

I see. I'd really like to avoid having to do that, especially since every developer would need to remember to do the same. Maybe there's just no good solution until https://github.com/rust-lang/cargo/issues/3670 has been fixed. I guess the problem is effectively the same as with using e.g. the `git` binary in tests. The difference is just that the binary can be built by `cargo`.

**Review by arxanas — APPROVED** [review]
Due to a recent build of `jj`, I'm now a proud owner of the `fake-editor` and `fake-diff-editor` binaries on my machine 😊

**Comment by martinvonz** [comment]
> Due to a recent build of `jj`, I'm now a proud owner of the `fake-editor` and `fake-diff-editor` binaries on my machine 😊

Sorry :( I don't know what the right solution to this problem is. I suppose we will also need a `git` binary on the path for tests in not too long (for running a git server to testing push/pull with, and to test GC, neither of which libgit2 supports AFAIK). For `git`, we can look for it on the path and have a `$TEST_GIT` override like I think git-branchless has. That doesn't work as well for these `fake-*` binaries, though.

File: tests/common/mod.rs [hunk 1]
```diff
@@ -140,7 +140,7 @@ impl TestEnvironment {
 
     /// Sets up the fake editor to read an edit script from the returned path
     pub fn set_up_fake_editor(&mut self) -> PathBuf {
-        let editor_path = assert_cmd::cargo::cargo_bin("fake-editor");
+        let editor_path = assert_cmd::cargo::cargo_bin("examples/fake-editor");
```
Comment (arxanas) [inline tests/common/mod.rs:143]: Interesting, this really works? How does Cargo know that it needs to build the example before running the test?

Comment (martinvonz) [reply tests/common/mod.rs:143]: Yep, it seems that it does (the checks pass). I think you can have regular tests in the files in `examples/`, so maybe `cargo test` builds them for that reason. But that's just my guess.

There's an unstable feature that lets us depend on crates with just binary targets (no library). So once that's in stable, we can switch to that setup. Hopefully this solution doesn't break before then :)

Comment (martinvonz) [reply tests/common/mod.rs:143]: Oh, it actually doesn't work :( It works with `cargo nextest run`, however, which is why I didn't notice that it was broken with regular `cargo test`. I don't know why the GitHub actions pass.

**Pull request closed without merging**